packages/fxa-settings/src/components/App/index.tsx

```diff
@@ -92,16 +99,23 @@ export const App = ({
 
   const isSignedIn = isSignedInData?.isSignedIn;
 
+  const getMetricsEnabled = useCallback(() => {
+    if (metricsLoading || !integration || isSignedIn === undefined) {
+      return;
+    }
+    return data?.account?.metricsEnabled || !isSignedIn;
```

Comment:
This logic looks identical to the [old line 104](https://github.com/mozilla/fxa/pull/16696/files#diff-36aa53053f1cbb57d4a94e4cf502a82d89c9ef2cfdb037ffc6e31ad298c74893L104) below, so I don't think I understand how this changes the handling of state change of `isSignedIn` while we still don't have a verified session to get the actual user opt-out pref.